torch/_inductor/utils.py

```diff
@@ -1538,6 +1538,15 @@ def num_fw_fixed_arguments(dynamo_gm_num_inputs: int, aot_fw_gm_num_inputs: int)
     num_rng_seed_offset_inputs = (
         2 if torch._functorch.config.functionalize_rng_ops else 0
     )
+    # AOT won't lift any parameters if we're inlining NN Modules
+    # however desugaring subclasses will still add arguments
+    # resulted in extra fixed inputs https://github.com/pytorch/pytorch/issues/130502
+    if (
+        torch._dynamo.config.inline_inbuilt_nn_modules
+        and not torch._dynamo.utils.is_parameter_freezing()
```

Comment:
hmm maybe i'm missing something - but with your PR above this one (plumbing `fw_metadata. static_input_indices` through to inductor), why do we still need this function?

Reply:
if we're parameter freezing we still lift some params as constants :(, and we don't inline for export either

Reply:
After some offline discussion, I think we will be able to remove this function once inlining is on by default.